src/main/java/com/team973/lib/util/Subsystem.java

```diff
@@ -1,20 +1,63 @@
 package com.team973.lib.util;
 
 /** Base interface for all subsystems */
-public interface Subsystem {
+public abstract class Subsystem<K extends Enum<K>> {
+  public abstract static class Stateless extends Subsystem<Stateless.FakeState> {
+    private final StateMap<FakeState> m_stateMap = new StateMap<>(FakeState.class);
+
+    public enum FakeState {
```

Comment:
Nit: rather than naming this FakeState, couldn't it just be named "State" with a single entry such as "SingleState" or similar?